src/Package/Impl/Shell/VsContextMenu.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Windows;
+using Microsoft.R.Components.Controller;
+using Microsoft.VisualStudio.R.Package.Interop;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.R.Package.Shell {
+    public static class VsContextMenu {
+        public static void Show(Guid guid, int id, ICommandTarget commandTarget, Point screenPosition) {
```

Comment:
This one goes via IvsUIShell that has ability to supply command target rather than have menu service send commands via VS chain. I guess we can add optional command target parameter to ICoreShell.ShowContextMenu and have it do it via menu service or via command target.